src/waveform/vsyncthread.cpp

```diff
@@ -63,7 +63,9 @@ void VSyncThread::run() {
                 m_timer.elapsed().toIntegerMicros());
             // waiting for interval by sleep
             if (remainingForSwap > 100) {
-                usleep(remainingForSwap);
+                // Leave a bit of margin for event delivery. In the worst case we will
+                // call swap a bit too early, so it will block, but not more than 100us.
+                usleep(remainingForSwap - 100);
```

Comment:
The reason for > 100 was that usleep() was not able to sleep shorter on my test device at that time. 
Not sure how the value is now. 

The -100 here should have no effect, because if we are to early, the timer will loaded with +100 to compensate this adjustment. I have just tested with 3000 also without any effect.  Does it have an effect on your system?

Reply:
I don't understand why you say subtracting 100 from the sleep time will have no effect. If I understand correctly, remainingForSwap is the time until the next swap is supposed to happen, so, if we call swap at that exact moment, the swap will take place immediately. If we would call swap a bit earlier, it will block until the swap takes place. My reason to sleep a bit less, and call swap a bit earlier, is that we are using a queued signal/slot to make the call. If the queued signal/slot takes to long, we will miss the swap and the call to swap will significantly longer. By sleeping a bit less, we reduce that risk. 

But the truth is, this is speculation. I will do some logging to check if this actually occurs.